### ISSUE TITLE
variadic + indexer bug

### ISSUE THREAD

**Issue by mohayonao** [babel/babel#917]
on 4.6.1

```js
let get1stArg = (...args) => args[0];

let a = get1stArg(1, 2, 3);

console.log(a);
```
expected: `1`
actual: `object`

compiled:
```js
var _arguments = arguments;
var get1stArg = function () {
  return _arguments[0];
};
```

I noticed that the below code runs correctly.

```js
let get1stArg = (...args) => args.slice()[0];
```

Maybe, this issue arises by directly accessing variadic with indexer.

**Comment by neVERberleRfellerER** [comment]
@mohayonao This is unforeseen consequence of my optimizations introduced in 4.6.0. We optimize literal subscript and expressions that evaluate to number to directly take item from `arguments`, so there is no need for copy of arguments. Since arguments are resolved to binding in lexically enclosing environment, it breaks.
Should be quick fix which I am going to do now.
I am sorry for these problems.

**Comment by sebmck** [comment]
Why do you expect an object? The first argument is 1 so this looks like
it's behaving correctly, can you elaborate?

On Sunday, 1 March 2015, mohayonao <notifications@github.com> wrote:

> on 4.6.1
>
> let get1stArg = (...args) => args[0];
> let a = get1stArg(1, 2, 3);
> console.log(a);
>
> expected: 1
> actual: object
>
> compiled:
>
> var _arguments = arguments;var get1stArg = function () {
>   return _arguments[0];
> };
>
> I noticed that the below code runs correctly.
>
> let get1stArg = (...args) => args.slice()[0];
>
> Maybe, this issue arises by directly accessing variadic with indexer.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/issues/917>.
>


-- 
Sebastian McKenzie

**Comment by neVERberleRfellerER** [comment]
@sebmck Sample code is incomplete, but the problem is, that arguments are remapped inside arrow function and optimized accesses are refering to _arguments instead of arguments

**Comment by sebmck** [comment]
Fixed as of 4.6.3, thanks!
